In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Importing Libraries
The required Python libraries were imported for data analysis and visualization. Pandas and NumPy are used for data handling, while Matplotlib and Seaborn are used to create visual charts.

In [3]:
dataset = pd.read_csv("UberDataset.csv")
dataset.head()

,START_DATE,END_DATE,CATEGORY,START,STOP,MILES,PURPOSE
0,01-01-2016 21:11,01-01-2016 21:17,Business,Fort Pierce,Fort Pierce,5.1,Meal/Entertain
1,01-02-2016 01:25,01-02-2016 01:37,Business,Fort Pierce,Fort Pierce,5.0,NaN
2,01-02-2016 20:25,01-02-2016 20:38,Business,Fort Pierce,Fort Pierce,4.8,Errand/Supplies
3,01-05-2016 17:31,01-05-2016 17:45,Business,Fort Pierce,Fort Pierce,4.7,Meeting
4,01-06-2016 14:42,01-06-2016 15:49,Business,Fort Pierce,West Palm Beach,63.7,Customer Visit


# Loading and Previewing the Dataset
The Uber trip dataset was loaded using read_csv(), and the first five rows were displayed using head(). This helps understand the structure of the dataset and view important columns such as start date, end date, category, locations, miles, and purpose.

In [4]:
dataset.shape

(1156, 7)

# Checking Dataset Shape
The shape function was used to check the size of the dataset. The output shows that the dataset contains 1,156 rows and 7 columns.

In [5]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1156 entries, 0 to 1155
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   START_DATE  1156 non-null   object 
 1   END_DATE    1155 non-null   object 
 2   CATEGORY    1155 non-null   object 
 3   START       1155 non-null   object 
 4   STOP        1155 non-null   object 
 5   MILES       1156 non-null   float64
 6   PURPOSE     653 non-null    object 
dtypes: float64(1), object(6)
memory usage: 63.3+ KB


# Checking Dataset Information
The info() function was used to review the column names, data types, and missing values. The output shows that most columns are stored as object data types, while MILES is numeric. It also shows missing values in columns such as CATEGORY, START, STOP, and especially PURPOSE.